mailbox/plugin/quota-search/src/main/java/org/apache/james/quota/search/QuotaQuery.java

```diff
@@ -0,0 +1,130 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.quota.search;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import org.apache.james.core.Domain;
+import org.apache.james.mailbox.model.QuotaThreshold;
+import org.apache.james.util.streams.Limit;
```

Comment:
it's not the Limit instance you want to use, when building a QuotaQuery, you shouldn't need to use something from org.apache.james.util.streams
Same thing about Offset.

Reply:
I do disagree with this.

IMO a Limit and an Offset are generic concepts.

Perhaps we need to remove specific stream knowledge in Limit, and relocate the package to better reflect this genericity.